src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -291,9 +307,55 @@ impl KvmVcpu {
             vcpu_events,
             xcrs,
             xsave,
+            tsc_khz,
         })
     }
 
+    // Scale the TSC frequency of this vCPU to the one provided as a parameter.
+    pub fn set_tsc_khz(&self, state_tsc: Option<u32>, cpuid: &CpuId) -> Result<()> {
+        // Check if CPU models are the same.
+        if let Ok(curr_cpuid) = cpuid::common::get_cpuid(0x1, 0) {
+            if is_same_model(cpuid, curr_cpuid.eax) {
```

Comment:
This allow snapshots to be loaded on the same CPU, even if there is no constant TSC or TSC scaling support. Are we sure the checks we do here guarantee that TSC freq is the same ?

Reply:
If we as a process never change the TSC frequency outside the scope of a snapshot save/restore, and the CPU model is the same, I expect the TSC frequency to be the same in the clone as in the original VM. If this is not true, then checking the CPU model is actually useless, since we still need to get/set the TSC frequency regardless.

On variant TSC systems, I don't think users even expect high precision from the TSC, but I might have a bad understanding of this whole story.